rust/arrow/src/datatypes.rs

```diff
@@ -946,10 +1055,46 @@ impl Field {
                     },
                     _ => data_type,
                 };
+
+                let mut dict_id = 0;
+                let mut dict_is_ordered = false;
+
+                let data_type = match map.get("dictionary") {
+                    Some(dictionary) => {
+                        let index_type = match dictionary.get("indexType") {
+                            Some(t) => DataType::from(t)?,
+                            _ => {
+                                return Err(ArrowError::ParseError(
+                                    "Field missing 'type' attribute".to_string(),
+                                ));
+                            }
+                        };
+                        dict_id = match dictionary.get("id") {
+                            Some(Value::Number(n)) => n.as_i64().unwrap(),
+                            _ => {
+                                return Err(ArrowError::ParseError(
+                                    "Field missing 'id' attribute".to_string(),
+                                ));
+                            }
+                        };
+                        dict_is_ordered = match dictionary.get("isOrdered") {
+                            Some(&Value::Bool(n)) => n,
+                            _ => {
+                                return Err(ArrowError::ParseError(
+                                    "Field missing 'id' attribute".to_string(),
```

Comment:
Error message refers to `id` instead of `isOrdered`

Reply:
Fixed in 2365cb3